antarest/study/storage/output_service.py

```diff
@@ -545,3 +549,36 @@ def aggregate_output_task(notifier: ITaskNotifier) -> TaskResult:
         )
 
         return download_id
+
+    def get_output_variables_metadata(self, study_id: str, output_id: str) -> OutputVariablesMetadata:
+        """
+        Returns metadata concerning a given output variables.
+        First, try to fetch the given data inside DB.
+        If present, return the data.
+        If not, parse the output headers to build the object. Before returning it, save it inside DB for next calls.
+        """
+        study = self._study_service.get_study(study_id)
+        assert_permission(study, StudyPermissionType.READ)
+
+        with db():
+            output_variables: OutputVariables | None = db.session.get(OutputVariables, (study_id, output_id))
+            if output_variables:
+                return output_variables.to_model()
+
+        # Fetches the data inside the FS
+        output_path = self._storage.get_output_path(study, output_id)
+        model = self._output_variables_manager.get_variables_metadata(output_path)
+
+        # Save the model inside DB for next calls
+        with db():
+            db_model = OutputVariables(
+                study_id=study_id,
+                output_id=output_id,
+                variables_metadata_version=1,
```

Comment:
This version number should not be hardcoded in my opinion, you can extract it as a constant. And also use this constant as a default value in the model